src/app/code/community/Divante/VueStorefrontIndexer/Model/Index/Mapping/Eav/Abstract.php

```diff
@@ -91,6 +101,18 @@ public function getAttributeType(Attribute $attribute)
     {
         $attributeCode = $attribute->getAttributeCode();
 
+        if ('visibility' === $attributeCode) {
+            return FieldInterface::TYPE_INT;
+        }
+
+        if ('status' === $attributeCode) {
+            return FieldInterface::TYPE_INT;
+        }
+
+        if (('select' === $attribute->getFrontendInput() || 'multiselect' === $attribute->getFrontendInput()) && 'eav/entity_attribute_source_boolean' !== $attribute->getSourceModel()) {
+            return FieldInterface::TYPE_TEXT;
+        }
```

Comment:
changes regarding 'select' and 'multiselect' can be excepted (it will not work with VSF core)
@pkarw

Reply:
Well this is also related to my previous comment about option_ids. If it is true that in EE this is transmitted as IDs instead of text, then this code is still relevant.

Reply:
Hmm, VSF is using integer values for multi select and select fields - using the [optionLabel](https://github.com/pkarw/vue-storefront/blob/726e21a0d58fd1dc73b15310a6b996d9c171df32/core/modules/catalog/helpers/optionLabel.ts#L10) helper to map the integer value for label. I'm not sure if it's handled differently in EE - but probably not. So I would love to keep the integers